### PR TITLE
Add update on `is_valid_char`

### PR DESCRIPTION
**Pull request by ScottPJones**
This is per #11171

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -178,6 +178,8 @@ Library improvements
     * `charwidth(c)` and `strwidth(s)` now return up-to-date cross-platform
       results (via utf8proc) ([#10659]): Julia now likes pizza ([#3721]), but some terminals still don't.
 
+    * `is_valid_char(c)` now correctly handles Unicode "non-characters", which are valid Unicode codepoints.
```
Comment (pao) [inline NEWS.md]: Cross reference the issue this fixes (or pull request) as you see in the above. There's a script somewhere (maybe in contrib? got an obvious name but I'm on my tablet and lazy) to run after which will generate the correct footnote.

Comment (ScottPJones) [inline NEWS.md]: Thanks so much! Again, you've been a big help.  Will do.

**Comment by pao** [comment]
p.s. you will need to run the script, too--sorry that wasn't clear

(someone else will have to merge this, I'm going to be out of touch for a couple weeks)

**Comment by ScottPJones** [comment]
Help!  What script? (sorry for more newbieness with Git, GitHub, etc!)

**Comment by kmsquire** [comment]
The script that @pao mentioned above is actually `doc/NEWS-update.jl` (found by typing `script update news` in the github search box for julia).  Cheers!

**Comment by ScottPJones** [comment]
@kmsquire Thanks, but I'm still a bit unclear about what I need to do (I created the PR directly on GitHub, so I don't even have a local branch).  Do I need to: 1) create and sync a local branch, 2) find and run the NEWS-update.jl script [I haven't used julia as a scripting language yet... will have to investigate that too] and then 3) push whatever changes the script made back to GitHub?

**Comment by kmsquire** [comment]
Pretty much, yes.

**Comment by ScottPJones** [comment]
Thanks again for all the help! Maybe some year, hopefully before I retire, I'll have all this figured out! :grinning:

**Comment by kmsquire** [comment]
Sure!  Since this is a one-line change, I would suggest just closing this request, making the change locally, running the update script, and submitting a new PR.

**Comment by stevengj** [comment]
(Note that, in general, we haven't been adding NEWS items for bug fixes.)

**Pull request merged**